lib/Sema/TypeCheckDecl.cpp

```diff
@@ -1373,6 +1373,43 @@ DefaultDefinitionTypeRequest::evaluate(Evaluator &evaluator,
   return Type();
 }
 
+llvm::Expected<bool>
+DidSetContainsOldValueReferenceRequest::evaluate(Evaluator &evaluator,
+                                                 AccessorDecl *decl) const {
+
+  class OldValueFinder : public ASTWalker {
+    const ASTContext &Ctx;
+    bool foundOldValueRef = false;
+
+  public:
+    OldValueFinder(const ASTContext &ctx) : Ctx(ctx) {}
+
+    virtual std::pair<bool, Expr *> walkToExprPre(Expr *E) override {
+      if (!E)
+        return {true, E};
+      if (auto DRE = dyn_cast<DeclRefExpr>(E)) {
+        if (DRE->getDecl()->getBaseName().getIdentifier() == Ctx.Id_oldValue) {
```

Comment:
Does this crash if the base name is a special name, eg `init`?

Reply:
Haven't tried it but I think it will, better to skip special names.